src/Products/sources/TestSource.js

```diff
@@ -2,8 +2,45 @@ import testData from './test-data.json';
 
 const TestSource = {
   read(name) {
+    //populate daysLeft property for each item
+    testData[name].forEach((element) => {
+      element.daysLeft = calculateDaysLeft(element.createdAt, element.coolingPeriod);
+    });
+
     return testData[name];
-  }
-}
+  },
+
+  write(name, entity, id) {
+    //try remove old version if exists
+    if (id) {
```

Comment:
If we have time to use LocalStorage as `DataSource` or if we decide to implement a backend service to persist the products, we will have to copy and paste this logic to `src/Products/sources/LocalStorageSource.js` for example. What do you think to move this logic to `/src/Products/index.js`? This way we leave the data sources with just the responsibility to persist the product, nothing else.

Reply:
Good point. I've moved some logic to index.js. Though, I think, what concerns id generation - it's data source responsibility. A lot of Web APIs and underlying databases return autogenerated ids, so I'd leave it here.  
eb77c82